vita3k/gui/src/settings_dialog.cpp

```diff
@@ -170,9 +179,13 @@ void draw_settings_dialog(GuiState &gui, HostState &host) {
             }
             ImGui::SameLine();
         } else {
-            ImGui::TextColored(GUI_COLOR_TEXT, "No modules present.\nPlease install the firmware");
-            if (ImGui::IsItemHovered())
-                ImGui::SetTooltip("Put decrypted modules inside 'vs0/sys/external/'.");
+            ImGui::TextColored(GUI_COLOR_TEXT, "No modules present.\nPlease download and install the last firmware.");
+            std::ostringstream link;
+            if (ImGui::Button("Download Firmware")) {
+                std::string firmware_url = "https://www.playstation.com/en-us/support/system-updates/ps-vita/";
+                link << OS_PREFIX << firmware_url;
+                system(link.str().c_str());
```

Comment:
AHHH `system`!

Zang pointed out in dms that this is already used in about_dialog.cpp. I guess it is okay.

I would've probably used `fmt::format("{} https://www.playstation.com/en-us/support/system-updates/ps-vita/", OS_PREFIX)` though.